### PR TITLE
4379 Feat: Add Code Analyzer

### PR DESCRIPTION
**Pull request by aaheli8**
What does this PR do?
Implements code analyzer.

Link your issue: #4379
Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
Yes! 👏

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
         "textalk/websocket": "1.5.2",
         "phpunit/phpunit": "^9.5.5",
         "vimeo/psalm": "^4.8.1",
-        "workerman/workerman": "^4.0"
+        "workerman/workerman": "^4.0",
+        "phpstan/phpstan": "^1.8"
```
Comment (stnguyen90) [inline composer.json:20]: It seems this file is missing the check script like https://github.com/utopia-php/ab/blob/0b386a8ff74199f1389daa54738523487a415d0f/composer.json#L13-L15

File: src/WebSocket/Adapter.php [hunk 2]
```diff
@@ -6,6 +6,11 @@ abstract class Adapter
 {
     protected string $host;
     protected int $port;
+      /**
+     
+     * @var array<int|string,bool|int|string>
+     */
+
```
Comment (stnguyen90) [inline src/WebSocket/Adapter.php:13]: Please remove the empty lines and fix the alignment

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
When I ran phpstan, I got 11 errors. Please make sure they're addressed:

```shell
$ docker run --rm -v $PWD:/app composer sh -c "composer install --profile --ignore-platform-reqs && ./vendor/bin/phpstan analyse --level max src tests"
[12.8MiB/0.27s] Installing dependencies from lock file (including require-dev)
[12.9MiB/0.27s] Verifying lock file contents can be installed on current platform.
[14.2MiB/0.28s] Nothing to install, update or remove
[14.0MiB/0.29s] Package webmozart/path-util is abandoned, you should avoid using it. Use symfony/filesystem instead.
[14.0MiB/0.29s] Generating autoload files
[14.4MiB/0.44s] 43 packages you are using are looking for funding.
[14.4MiB/0.44s] Use the `composer fund` command to find out more!
[14.4MiB/0.44s] Memory usage: 14.42MiB (peak: 15.89MiB), time: 0.44s
 7/7 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ----------------------------------------------------------------------- 
  Line   src/WebSocket/Adapter/Workerman.php                                    
 ------ ----------------------------------------------------------------------- 
  32     Parameter #1 $callback of function call_user_func expects callable():  
         mixed, mixed given.                                                    
  69     Cannot access property $onWebSocketConnect on mixed.                   
 ------ ----------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------- 
  Line   src/WebSocket/Server.php                                               
 ------ ----------------------------------------------------------------------- 
  17     Property Utopia\WebSocket\Server::$errorCallbacks type has no value    
         type specified in iterable type array.                                 
         💡 See:                                                                
         https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-i  
         terable-type                                                           
 ------ ----------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------- 
  Line   tests/servers/Swoole/server.php                                      
 ------ --------------------------------------------------------------------- 
  28     Parameter #1 $connections of method Utopia\WebSocket\Server::send()  
         expects array<string>, array<int, int> given.                        
  31     Parameter #1 $connections of method Utopia\WebSocket\Server::send()  
         expects array<string>, array<int, int> given.                        
  34     Parameter #1 $connections of method Utopia\WebSocket\Server::send()  
         expects array<string>, array given.                                  
  37     Parameter #1 $connections of method Utopia\WebSocket\Server::send()  
         expects array<string>, array<int, int> given.                        
 ------ --------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------- 
  Line   tests/servers/Workerman/server.php                                   
 ------ --------------------------------------------------------------------- 
  27     Parameter #1 $connections of method Utopia\WebSocket\Server::send()  
         expects array<string>, array<int, int> given.                        
  30     Parameter #1 $connections of method Utopia\WebSocket\Server::send()  
         expects array<string>, array<int, int> given.                        
  33     Parameter #1 $connections of method Utopia\WebSocket\Server::send()  
         expects array<string>, array given.                                  
  36     Parameter #1 $connections of method Utopia\WebSocket\Server::send()  
         expects array<string>, array<int, int> given.                        
 ------ --------------------------------------------------------------------- 

 [ERROR] Found 11 errors  
 ```

File: src/WebSocket/Adapter.php (hunk 2 above)
Comment (aaheli8) [reply src/WebSocket/Adapter.php:13]: Working on resolving the errors.

**Comment by aaheli8** [comment]
I have resolved the errors as mentioned but I am unable to resolve one particular error and need help regarding it.
image
Please provide some help to solve it. Thank you.

File: composer.json (hunk 1 above)
Comment (aaheli8) [reply composer.json:23]: I have resolved this issue

**Comment by gewenyu99** [comment]
@stnguyen90 Teehee Please look at this again.

**Comment by aaheli8** [comment]
Resolved all the errors. Thank you @stnguyen90 for helping to resolve the last error.
image

**Comment by aaheli8** [comment]
Would it take more time for review, I am asking because today is the last for hacktoberfest.

**Comment by PineappleIOnic** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

**Review by stnguyen90 — APPROVED** [review]
Hey 👋 awesome work on your PR! We've approved your work and it'll be merged soon! 

I know there's a build error, but it looks like that was already on the repo. We'll try to fix it in another PR.

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**